### PR TITLE
Add sample text to angular calculator component.

### PR DESCRIPTION
**Pull request by roeerokah**
Add DomTestingService, provideMock and Spied<T>

### CODE REVIEW

File: Angular/angular.json [hunk 1]
```diff
@@ -100,5 +100,8 @@
         }
       }
     }
+  },
+  "cli": {
+    "analytics": false
   }
```
Comment (roeerokah) [inline Angular/angular.json:106]: Will not ask them if they want to log analytics to angular

File: Angular/src/app/calculator/calculator.component.html [hunk 2]
```diff
@@ -1,28 +1,28 @@
 <div class="calculator" [formGroup]="calculatorForm">
-  <h1 class="calculator__title">Calculator</h1>
+  <h1 class="calculator-title" data-test="calculator-title">Calculator</h1>
```
Comment (roeerokah) [inline Angular/src/app/calculator/calculator.component.html:2]: changed the class names to be on our convention

File: Angular/src/app/calculator/calculator.component.spec.ts [hunk 3]
```diff
@@ -1,24 +1,56 @@
...
+import {provideMock, Spied} from "../helpers/provide-mock";
+
+
+const dataTests = {
+  title: '[data-test="calculator-title"]',
+  firstNumberInput: '[data-test="first-number-input"]',
+  secondNumberInput: '[data-test="second-number-input"]',
+  calculateButton: '[data-test="calculate-button"]',
+  result: '[data-test="result"]',
+};
```
Comment (roeerokah) [inline Angular/src/app/calculator/calculator.component.spec.ts:16]: Added our tests conventions, also added DomTestingService(helps to interact with the DOM) and provideMock (which mocks a service), it's from our testing conversions

File: Angular/src/app/calculator/calculator.component.spec.ts [hunk 4]
```diff
@@ -1,24 +1,56 @@
...
+  describe('Component finished rendering', () => {
+    beforeEach(() => {
+      fixture.detectChanges();
+    });
+
+    it('should show title', () => {
+      expect(domTestingService.getElement(dataTests.title)).toBeTruthy();
+    });
+
+    it('first number input should be 0', () => {
```
Comment (OrMenashe) [inline Angular/src/app/calculator/calculator.component.spec.ts]: can you change it to `"first number input should be the default"`,
inside the test, create variable `expectedDefaultNumber` and use it in the expectations,
same for the other tests

File: Angular/src/app/helpers/dom-testing.service.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+export class DomTestingService {
```
Comment (OrMenashe) [inline Angular/src/app/helpers/dom-testing.service.ts:32]: can you explain what is this?

File: Angular/src/app/helpers/provide-mock.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import {Type} from "@angular/core";
+
+export type Spied<T> = {
+  [Method in keyof T]: any;
+};
+
+export function spyOnClass(spiedClass: Type<any>) {
```
Comment (OrMenashe) [inline Angular/src/app/helpers/provide-mock.ts:7]: can you explain what is this?

File: Angular/src/app/helpers/provide-mock.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      return "value" in descriptor ? descriptor.value instanceof Function : false;
+    })
+    .map(([name]) => name);
+  const getters = propertiesDescriptors
+    .filter(([, descriptor]) => {
+      // @ts-ignore
+      return !!descriptor.get;
+    })
+    .map(([name]) => name);
+  return jasmine.createSpyObj('spy', [...methods, ...getters]);
```
Comment (OrMenashe) [inline Angular/src/app/helpers/provide-mock.ts:22]: in our course we are going to use jest and not jasmine

**Review by OrMenashe — APPROVED** [review]

**Pull request merged**